contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -198,8 +198,12 @@ def checkstyle(self, interpreter, sources):
                                      labels=[WorkUnitLabel.TOOL, WorkUnitLabel.LINT],
                                      cmd=' '.join(checker.cmdline(args))) as workunit:
         return checker.run(args=args,
-                                    stdout=workunit.output('stdout'),
-                                    stderr=workunit.output('stderr'))
+                           stdout=workunit.output('stdout'),
+                           stderr=workunit.output('stderr'),
+                           # Overriding PEX_PYTHON_PATH is necessary to satisfy the interpreter
+                           # constraint we added when creating the checker pex to guard against a
+                           # pexrc.
+                           env={'PEX_PYTHON_PATH': interpreter.binary})
```

Comment:
IIUC this only guards against a pexrc that sets PEX_PYTHON_PATH. It does not guard againts a pexrc that sets PEX_PYTHON (which is more appropriate to set here since we know the exact single interpreter we want). If so, I think the correct answer is to eliminate the `pex_builder.add_interpreter_constraint(str(interpreter.identity.requirement))` above and set the env up here to be:
```python
{'PEX_PYTHON': interpreter.binary,
 'PEX_IGNORE_RCFILES': 'True'}
```

Reply:
Oh, that sounds much more reasonable. I delved just enough into the pex source to answer the specific question without looking at the more appropriate one. Will do, thanks!

Reply:
Circling back to note this probably was just whac-a-mole. Something like the `PEX_IGNORE_RCFILES` suggested above is needed to fix this class of issues generally. #7816 points out what is likely the same problem with mypy and v2 also has this issue generally right now.